db/package.js

```diff
@@ -102,61 +102,96 @@ class PackageDbController {
   }
 
   async refreshOwnership ({ packages, registry, maintainerId }) {
+    // find all the packages in the DB that either are marked as maintained by me,
+    // or simply have the provided registry and a name that's in the list provided by the registry
     const existingPackages = await this.db.collection('packages').find({
       $or: [
         { name: { $in: packages } },
-        { owner: maintainerId }
+        { 'maintainers.maintainerId': maintainerId }
+        // https://docs.mongodb.com/manual/tutorial/query-array-of-documents/#specify-a-query-condition-on-a-field-in-an-array-of-documents
       ],
       registry
     }).toArray()
 
+    console.error('found existing pkgs for this maintainer', existingPackages)
+
+    // of the packages already marked as maintained by me, whichever aren't in the list of
+    // packages provided, remove my id from their maintainers list
     const packageDeletions = existingPackages
       .filter(pkg => !packages.includes(pkg.name))
       .map(pkg => ({
-        criteria: { name: pkg.name, registry },
+        criteria: { _id: pkg._id },
         update: {
-          $set: { owner: null },
           $pull: { maintainers: { maintainerId } }
         }
       }))
+
+    // of the packages provided, whichever aren't already marked as maintained by me,
```

Comment:
is this more about "of the packages provided, if any aren't created in our DB yet, create them with me as a maintainer (also should be % 100 if that's the case)

Reply:
i think you're right; will update it